tests/unit/s2n_server_hello_retry_test.c

```diff
@@ -56,6 +96,82 @@ static int client_hello_detect_duplicate_calls(struct s2n_connection *conn, void
     return 0;
 }
 
+int s2n_client_hello_poll_cb(struct s2n_connection *conn, void *ctx)
+{
+    struct client_hello_context *client_hello_ctx;
+    if (ctx == NULL) {
+        return -1;
+    }
+    client_hello_ctx = ctx;
+    /* Increment counter to ensure that callback was invoked */
+    client_hello_ctx->invocations++;
+
+    if (client_hello_ctx->mark_done) {
+        EXPECT_SUCCESS(s2n_client_hello_cb_done(conn));
+        return S2N_SUCCESS;
+    }
+
+    return S2N_SUCCESS;
+}
+
+S2N_RESULT hello_retry_client_hello_cb_test(bool enable_poll) {
+    DEFER_CLEANUP(struct s2n_config *client_config = s2n_config_new(), s2n_config_ptr_free);
+    EXPECT_NOT_NULL(client_config);
+    DEFER_CLEANUP(struct s2n_config *server_config = s2n_config_new(), s2n_config_ptr_free);
+    EXPECT_NOT_NULL(server_config);
+
+    DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+    DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
+    EXPECT_NOT_NULL(server_conn);
+    EXPECT_NOT_NULL(client_conn);
+
+    struct s2n_cert_chain_and_key *tls13_chain_and_key;
+    EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&tls13_chain_and_key,
+        S2N_ECDSA_P384_PKCS1_CERT_CHAIN, S2N_ECDSA_P384_PKCS1_KEY));
+    EXPECT_NOT_NULL(tls13_chain_and_key);
+
+    EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(server_config));
+    EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(client_config));
+
+    EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(client_config, tls13_chain_and_key));
+    EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, tls13_chain_and_key));
+
+    EXPECT_SUCCESS(s2n_connection_set_config(server_conn, server_config));
+    EXPECT_SUCCESS(s2n_connection_set_config(client_conn, client_config));
```

Comment:
Nit: My point was that this is unnecessarily verbose / complicated. You only need one config if your configs are identical. Creating two makes the reader think they must be different in some way.
```suggestion
S2N_RESULT hello_retry_client_hello_cb_test(bool enable_poll) {
    struct s2n_cert_chain_and_key *tls13_chain_and_key = NULL;
    EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&tls13_chain_and_key,
        S2N_ECDSA_P384_PKCS1_CERT_CHAIN, S2N_ECDSA_P384_PKCS1_KEY));
    EXPECT_NOT_NULL(tls13_chain_and_key);

    DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
    EXPECT_NOT_NULL(config);
    EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(config));
    EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, tls13_chain_and_key));
    EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default_tls13"));

    DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
    DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
    EXPECT_NOT_NULL(server_conn);
    EXPECT_NOT_NULL(client_conn);
    EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
    EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
```